sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala

```diff
@@ -164,6 +164,10 @@ class CSVOptions(
       s"${DateFormatter.defaultPattern}'T'HH:mm:ss[.SSS][XXX]"
     })
 
+  val timestampNTZFormatInRead: Option[String] = parameters.get("timestampNTZFormat")
+  val timestampNTZFormatInWrite: String = parameters.getOrElse("timestampNTZFormat",
+    s"${DateFormatter.defaultPattern}'T'HH:mm:ss[.SSS]")
```

Comment:
I wonder what's the reason to have the optional field `[.SSS]` in write. How should CSV writer decide whether to write milliseconds or not?

Another question, why the precision is in milliseconds but not in microseconds?

Reply:
> It is the same reason as for timestampFormat above ...

The option was added when Spark's timestamp type had milliseconds precision. Now the precision is microsecond, and don't see any reasons to loose info in write, by default.

Reply:
I think we can change it for both LTZ and NTZ, if the precision lose here is a problem.

Reply:
I would rather change it separately, sounds more like a general problem with timestamps.